### PR TITLE
🐛 Fix for integration test cases with old Decision Point keys & tests for Enrollment Codes 

### LINKED ISSUE #1407: ✨ Integration test case utils function checkExperimentAssignedIsNull() has old decision point keys {expId, expPoint} object check
**Version where bug was found:**
e.g "5.1"

**Describe the bug**
Integration test case utililty function `checkExperimentAssignedIsNull()` has old decision point keys {expId, expPoint} in object check. We should use {target, site}. This will affect the exclusion checks for various test cases as all were passing this function as this function uses != object [comparison](https://github.com/CarnegieLearningWeb/UpGrade/blob/27367f575ccdc3917df43f893de74084b0868c86/backend/packages/Upgrade/test/integration/utils/index.ts#L17).

### PR DESCRIPTION
**Pull request by ppratikcr7**
This PR resolves #1407 

The test cases are updated mainly due to the old Decision Point keys { expId, expPoint } used in the checkExperimentAssignedIsNull() function instead of { target, site }

There were 3 more test cases related to Enrollment codes that were commented, which are now revived with proper enrollment code checks.  

We would be building the exclusions code test cases on top of these test cases.

### CODE REVIEW

File: backend/packages/Upgrade/test/integration/Experiment/enrollmentCode/ExperimentExperimentEnrollmentCode.ts [hunk 1]
```diff
@@ -1,259 +1,269 @@
...
+export default async function testCase(): Promise<void> {
+  const experimentService = Container.get<ExperimentService>(ExperimentService);
+  const userService = Container.get<UserService>(UserService);
+
+  // creating new user
+  const user = await userService.upsertUser(systemUser as any, new UpgradeLogger());
+
+  // experiment object
+  const experimentObject = individualAssignmentExperimentConsistencyRuleExperiment;
+  experimentObject.partitions = [
```
Comment (VivekFitkariwala) [inline backend/packages/Upgrade/test/integration/Experiment/enrollmentCode/ExperimentExperimentEnrollmentCode.ts:24]: This is the same partition coming from the `individualAssignmentExperimentConsistencyRuleExperiment`. Why are you adding it again?

File: backend/packages/Upgrade/test/integration/ExperimentStats/DetailGroupExperiment.ts [hunk 2]
```diff
@@ -185,8 +185,6 @@ export default async function testCase(): Promise<void> {
   // user 2 logs in experiment
   // get all experiment condition for user 2
   experimentConditionAssignments = await getAllExperimentCondition(experimentUsers[1].id, new UpgradeLogger());
-  checkExperimentAssignedIsNull(experimentConditionAssignments, experimentName1, experimentPoint1);
```
Comment (VivekFitkariwala) [inline backend/packages/Upgrade/test/integration/ExperimentStats/DetailGroupExperiment.ts:188]: This function was checking no assignment. How is that getting checked now?

**Review by VivekFitkariwala — CHANGES_REQUESTED** [review]
I have added a few comments.

File: backend/packages/Upgrade/test/integration/Experiment/enrollmentCode/ExperimentExperimentEnrollmentCode.ts (hunk 1 above)
Comment (ppratikcr7) [reply backend/packages/Upgrade/test/integration/Experiment/enrollmentCode/ExperimentExperimentEnrollmentCode.ts:24]: I am setting excludeIfReached as true for these test cases.

File: backend/packages/Upgrade/test/integration/ExperimentStats/DetailGroupExperiment.ts (hunk 2 above)
Comment (ppratikcr7) [reply backend/packages/Upgrade/test/integration/ExperimentStats/DetailGroupExperiment.ts:188]: Yes. So, this `checkExperimentAssignedIsNull` function was not working as expected due to not matching to the correct DP keys. This function was not supposed to be called for the user2 as this user is not excluded. Instead we can update this to check no assignment for user1, as this user came before enrolling state and would be excluded.

File: backend/packages/Upgrade/test/integration/ExperimentStats/GroupEnrollment.ts [hunk 3]
```diff
@@ -93,8 +93,6 @@ export default async function testCase(): Promise<void> {
   // user 2 logs in experiment
   // get all experiment condition for user 2
   experimentConditionAssignments = await getAllExperimentCondition(experimentUsers[1].id, new UpgradeLogger());
-  checkExperimentAssignedIsNull(experimentConditionAssignments, experimentName1, experimentPoint1);
```
Comment (VivekFitkariwala) [inline backend/packages/Upgrade/test/integration/ExperimentStats/GroupEnrollment.ts:96]: Check that the assignments are not null

File: backend/packages/Upgrade/test/integration/ExperimentStats/IndividualEnrollment.ts [hunk 4]
```diff
@@ -29,7 +29,28 @@ export default async function testCase(): Promise<void> {
   const user = await userService.upsertUser(systemUser as any, new UpgradeLogger());
   // experiment object
   const experimentObject = individualExperimentStats;
-
+  experimentObject.partitions = [
```
Comment (VivekFitkariwala) [inline backend/packages/Upgrade/test/integration/ExperimentStats/IndividualEnrollment.ts:32]: Make a util function to update the flag

File: backend/packages/Upgrade/test/integration/ExperimentStats/IndividualEnrollment.ts [hunk 5]
```diff
@@ -115,29 +136,7 @@ export default async function testCase(): Promise<void> {
     ])
   );
 
-  experimentConditionAssignments = await getAllExperimentCondition(experimentUsers[0].id, new UpgradeLogger());
```
Comment (VivekFitkariwala) [inline backend/packages/Upgrade/test/integration/ExperimentStats/IndividualEnrollment.ts:118]: Why remove this code? It is checking that if the user comes back again it won't get any assignment. By removing this code the test case is no longer checking that scenario. You did this in many test cases please fix them

File: backend/packages/Upgrade/test/integration/ExperimentStats/WithinSubjectEnrollment.ts [hunk 6]
```diff
@@ -29,7 +29,28 @@ export default async function testCase(): Promise<void> {
   const user = await userService.upsertUser(systemUser as any, new UpgradeLogger());
   // experiment object
   const experimentObject = withinSubjectExperiment;
-
+  experimentObject.partitions = [
```
Comment (VivekFitkariwala) [inline backend/packages/Upgrade/test/integration/ExperimentStats/WithinSubjectEnrollment.ts:32]: Use utils function

File: backend/packages/Upgrade/test/integration/mockData/experiment/index.ts [hunk 7]
```diff
@@ -135,6 +135,62 @@ export const withinSubjectExperiment = {
   consistencyRule: null,
 };
 
+export const individualLevelEnrollmentCodeExperiment = {
```
Comment (VivekFitkariwala) [inline backend/packages/Upgrade/test/integration/mockData/experiment/index.ts:138]: Use util function

**Review by VivekFitkariwala — CHANGES_REQUESTED** [review]

File: backend/packages/Upgrade/test/integration/ExperimentStats/DetailGroupExperiment.ts (hunk 2 above)
Comment (ppratikcr7) [reply backend/packages/Upgrade/test/integration/ExperimentStats/DetailGroupExperiment.ts:188]: Okay, we use the function `checkExperimentAssignedIsNotDefault` to check if the assigned condition is not default condition everywhere in test cases. Will add that.

File: backend/packages/Upgrade/test/integration/Experiment/conditionalStateChange/GroupUserCount.ts [hunk 8]
```diff
@@ -229,13 +208,6 @@ export default async function GroupUserCount(): Promise<void> {
   );
   checkMarkExperimentPointForUser(markedExperimentPoint, experimentUsers[4].id, experimentName, experimentPoint);
 
-  // get the number of group satisfied as 2
```
[the remark this answers is not in the document]
Comment (ppratikcr7) [reply backend/packages/Upgrade/test/integration/Experiment/conditionalStateChange/GroupUserCount.ts:232]: This is getting checked at the end, there was a change in number of groups satisfied so this check is removed from here and is checked at the end of the test case.

File: backend/packages/Upgrade/test/integration/ExperimentStats/IndividualEnrollment.ts (hunk 5 above)
Comment (ppratikcr7) [reply backend/packages/Upgrade/test/integration/ExperimentStats/IndividualEnrollment.ts:118]: Okay, added this.

**Comment by ppratikcr7** [comment]
@VivekFitkariwala Can you please re-review the PR. I have addressed all of your comments in the last commit.

**Pull request merged**